presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/IterativeOptimizer.java

```diff
@@ -210,6 +211,45 @@ private boolean exploreNode(int group, Context context, Matcher matcher)
         return progress;
     }
 
+    private String getNameOfOptimizerRule(Rule<?> rule)
+    {
+        String ruleName = rule.getClass().getSimpleName();
+        if (rule instanceof RowExpressionRewriteRuleSet.ProjectRowExpressionRewrite) {
+            ruleName = ((RowExpressionRewriteRuleSet.ProjectRowExpressionRewrite) rule).getOptimizerNameForLog();
+        }
+        else if (rule instanceof RowExpressionRewriteRuleSet.SpatialJoinRowExpressionRewrite) {
+            ruleName = ((RowExpressionRewriteRuleSet.SpatialJoinRowExpressionRewrite) rule).getOptimizerNameForLog();
+        }
+        else if (rule instanceof RowExpressionRewriteRuleSet.JoinRowExpressionRewrite) {
+            ruleName = ((RowExpressionRewriteRuleSet.JoinRowExpressionRewrite) rule).getOptimizerNameForLog();
+        }
+        else if (rule instanceof RowExpressionRewriteRuleSet.JoinRowExpressionRewrite) {
+            ruleName = ((RowExpressionRewriteRuleSet.JoinRowExpressionRewrite) rule).getOptimizerNameForLog();
+        }
+        else if (rule instanceof RowExpressionRewriteRuleSet.WindowRowExpressionRewrite) {
+            ruleName = ((RowExpressionRewriteRuleSet.WindowRowExpressionRewrite) rule).getOptimizerNameForLog();
+        }
+        else if (rule instanceof RowExpressionRewriteRuleSet.ApplyRowExpressionRewrite) {
+            ruleName = ((RowExpressionRewriteRuleSet.ApplyRowExpressionRewrite) rule).getOptimizerNameForLog();
+        }
+        else if (rule instanceof RowExpressionRewriteRuleSet.FilterRowExpressionRewrite) {
+            ruleName = ((RowExpressionRewriteRuleSet.FilterRowExpressionRewrite) rule).getOptimizerNameForLog();
+        }
+        else if (rule instanceof RowExpressionRewriteRuleSet.ValuesRowExpressionRewrite) {
+            ruleName = ((RowExpressionRewriteRuleSet.ValuesRowExpressionRewrite) rule).getOptimizerNameForLog();
+        }
+        else if (rule instanceof RowExpressionRewriteRuleSet.AggregationRowExpressionRewrite) {
+            ruleName = ((RowExpressionRewriteRuleSet.AggregationRowExpressionRewrite) rule).getOptimizerNameForLog();
+        }
+        else if (rule instanceof RowExpressionRewriteRuleSet.TableFinishRowExpressionRewrite) {
+            ruleName = ((RowExpressionRewriteRuleSet.TableFinishRowExpressionRewrite) rule).getOptimizerNameForLog();
+        }
+        else if (rule instanceof RowExpressionRewriteRuleSet.TableWriterRowExpressionRewrite) {
```

Comment:
On adding a new rule you will have to extend this, wondering if there was a better way

Reply:
Yeah, I was trying to find a better way to achieve this but cannot think of one. Open to suggestions if there are better ways.

Reply:
Would the following patch work for you @feilong-liu ? I think you can just extend the rule interface to add a method which does this, and extend all of the rules using the class with the method instead, and check if it is an instance of the abstract class e.g.

```java
    public abstract class RowExpressionRewriteRule<T> implements Rule<T> {
        public String getOptimizerNameForLog() {
            String rewriterName = rewriter.getClass().getName();
            return format("%s:%s", rewriterName.substring(rewriterName.lastIndexOf('.') + 1), this.getClass().getSimpleName());
        }
    }

// rule definition
public final class ProjectRowExpressionRewrite
            extends RowExpressionRewriteRule<ProjectNode>
...
// name check 
        if (rule instanceof RowExpressionRewriteRuleSet.RowExpressionRewriteRule) {
            ruleName = ((RowExpressionRewriteRuleSet.RowExpressionRewriteRule) rule).getOptimizerNameForLog();
        }
```

[simplify-rulename.patch](https://github.com/prestodb/presto/files/15343575/simplify-rulename.patch)

Reply:
Good suggestion. Applied the change.